src/blueprint/azext_blueprint/_params.py

```diff
@@ -0,0 +1,222 @@
+
+    with self.argument_context('blueprint update') as c:
+        c.argument('blueprint_name', options_list=['--name', '-n'], help='Name of the blueprint definition.')
+        c.argument('description', help='Multi-line explain this resource.')
+        c.argument('parameters', arg_type=parameter_type, help='Parameters required by this blueprint definition.')
+
+    with self.argument_context('blueprint delete') as c:
+        c.argument('blueprint_name', options_list=['--name', '-n'], help='Name of the blueprint definition.')
+
+    with self.argument_context('blueprint show') as c:
+        c.argument('blueprint_name', options_list=['--name', '-n'], help='Name of the blueprint definition.')
+
+    with self.argument_context('blueprint list') as c:
+        pass
+
+    with self.argument_context('blueprint artifact delete') as c:
+        c.argument('blueprint_name', help='Name of the blueprint definition.')
+        c.argument('artifact_name', options_list=['--name', '-n'], help='Name of the blueprint artifact.')
+
+    with self.argument_context('blueprint artifact show') as c:
+        c.argument('blueprint_name', help='Name of the blueprint definition.')
+        c.argument('artifact_name', options_list=['--name', '-n'], help='Name of the blueprint artifact.')
+
+    with self.argument_context('blueprint artifact list') as c:
+        c.argument('blueprint_name', help='Name of the blueprint definition.')
+
+    with self.argument_context('blueprint resource-group add') as c:
+        c.argument('blueprint_name', help='Name of the blueprint definition.')
+        c.argument('rg_name', help='Name of this resource group. Leave empty if the resource group name will be specified during the blueprint assignment.')
+        c.argument('rg_location', help='Location of this resource group. Leave empty if the resource group location will be specified during the blueprint assignment.')
+        c.argument('artifact_name', help='A unique name of this resource group artifact.')
+        c.argument('display_name', help='Display name of this resource group artifact.')
+        c.argument('description', help='Description of the blueprint artifact.')
+        c.argument('depends_on', nargs='+', help='Artifacts which need to be deployed before the specified artifact.')
+        c.argument('tags', tags_type, help='Tags to be assigned to this resource group.')
+
+    with self.argument_context('blueprint resource-group update') as c:
+        c.argument('blueprint_name', help='Name of the blueprint definition.')
+        c.argument('rg_name', help='Name of this resource group. Leave empty if the resource group name will be specified during the blueprint assignment.')
+        c.argument('rg_location', help='Location of this resource group. Leave empty if the resource group location will be specified during the blueprint assignment.')

```

Comment:
just to confirm, location is updatable?

Reply:
Yes, you can update name and location of the resource group artifact in a blueprint.